src/TinyContainerBuilder.php

```diff
@@ -167,4 +167,43 @@ private function isAbsolute(string $path)
     {
         return (new Filesystem())->isAbsolutePath($path);
     }
+
+    public function excludeSourcePath(string $excludePath): ContainerBuilderInterface
+    {
+        if (!$this->isAbsolute($excludePath)) {
+            if (!isset($this->rootPath)) {
+                throw new \LogicException(
+                    \sprintf('When relative path is provided root should be set first. Privided: %s', $excludePath)
+                );
+            }
+            $excludePath = rtrim($this->rootPath, '/') . '/' . $excludePath;
+        }
+        if (!\file_exists($excludePath)) {
+            throw new \RuntimeException(\sprintf('Provided exclude path is not a valid pathname: %s', $excludePath));
+        }
+
+        // The extra slash at the end prevents exclusion of sibling paths starting with exclude path name
+        // For example /usr/bin/php shouldn't exclude /usr/bin/php7.4
+        $excludePath = $this->removeRelativePathParts($excludePath) . '/';
+        $this->paths = array_filter($this->paths, function (string $path) use ($excludePath) {
+            return 0 !== stripos($this->removeRelativePathParts($path) . '/', $excludePath);
+        });
+
+        return $this;
+    }
+
+    private function removeRelativePathParts(string $path): string
```

Comment:
though now that i think about it, i feel like the name is a bit misleading. "remove relative path parts" could be interpreted as taking "a/b/../c/d" and coming out with "a/b/c/d". 
Of course, since we will have it (correctly) come out as "a/c/d", we're not simply removing the relative parts.
Maybe this should be called something like `convertRelativePathToAbsolutePath` or the like?

Reply:
I renamed it to **resolve**RelativePathParts.